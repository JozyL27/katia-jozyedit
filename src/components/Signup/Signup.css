.Signup__container {
  width: 70%;
  font-size: 1.6rem;
  text-align: center;
  padding: 2rem 1rem 2rem;
}

.Signup__header {
  margin: 0 0 1rem;
  font-size: 2.75rem;
}

.Signup__form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.Signup__submit-button-div {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1rem 0 0;
  justify-content: center;
}

.Signup__or {
  font-size: 2rem;
  margin: .75rem 0;
}

.Signup__login-link {
  width: 7.5rem;
  border: none;
  background: none;
  font-weight: bolder;
  color: rgb(48, 100, 153);
  cursor: pointer;
  font-size: 1.8rem;
  transition-duration: 0.5ms;
}

#Signup__email, #Signup__display_name, 
#Signup__password, #Signup__confirm_password {
  border-radius: 0;
  -webkit-border-radius:0px;
}