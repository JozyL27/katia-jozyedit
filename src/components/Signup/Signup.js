import React, { Component } from 'react'
import {Link} from 'react-router-dom'
import TokenService from '../../services/token-service'
import AuthService from '../../services/auth-api-service'
import './Signup.css'

export default class Signup extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { email, display_name, password, confirm_password } = ev.target
    const user = {
      email: email.value.toLowerCase(),
      display_name: display_name.value,
      password: password.value
    }

    if(password.value === confirm_password.value) {
    AuthService.registerUser(user)
      .then(() => {
        AuthService.postLogin({
          email: email.value.toLowerCase().trim(),
          password: password.value
        })
          .then(res => {
            email.value = ''
            display_name.value = ''
            password.value = ''
            TokenService.saveAuthToken(res.authToken)
            this.props.onRegistrationSuccess(user.user_name)
          })
          .catch(error => this.setState({error: error.message}))
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
    } else if(password.value !== confirm_password.value) {
      this.setState({ error: 'passwords did not match' })
    }
  }

  render() {
    const { error } = this.state
    return (
      <>
      <main id='lp__main-flex'>
        <div className='lp__logoContainer'>
        <Link to='/' className='lp__lp-link'>Katia</Link>
        </div>
        <section className='Signup__container'>
          <h2 className='Signup__header'>Signup</h2>
          <form
            className='Signup__form'
            onSubmit={this.handleSubmit}
          >
            <div role='alert'>
              {error && <p className='Signup__error'>{error}</p>}
            </div>
            <div className='Signup__email-div lp__input-div'>
              <label className='lp__label' htmlFor='Signup__email'>
                Email
              </label>
              <input
                name='email'
                type='email'
                required
                id='Signup__email'
                className='lp__input'>
              </input>
            </div>
            <div className='Signup__display_name-div lp__input-div'>
              <label className='lp__label' htmlFor='Signup__display_name'>
                Display Name
              </label>
              <input
                name='display_name'
                type='text'
                required
                id='Signup__display_name'
                className='lp__input'>
              </input>
            </div>
            <div className='Signup__password-div lp__input-div'>
              <label className='lp__label' htmlFor='Signup__password'>
                Password 
              </label>
              <input
                name='password'
                type='password'
                required
                id='Signup__password'
                className='lp__input'>
              </input>
            </div>
            <div className='Signup__confirm_password-div lp__input-div'>
              <label className='lp__label' htmlFor='Signup__confirm_password'>
                Confirm Password 
              </label>
              <input
                name='confirm_password'
                type='password'
                required
                id='Signup__confirm_password'
                className='lp__input'>
              </input>
            </div>
            <button className='blue-button blue-button-margin' type='submit'>
              Submit
            </button>
            <div className='Signup__submit-button-div'>
              <p className='Signup__or'>Already have an account? </p>
              <Link className='Signup__login-link' to='/login'>Login</Link>
            </div>
          </form>
        </section>
      </main>
      </>
    )
  }
}

