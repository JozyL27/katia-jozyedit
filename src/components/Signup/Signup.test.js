import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Signup from './Signup'

describe('<Signup />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MemoryRouter>
        <Signup />
      </MemoryRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Signup />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})