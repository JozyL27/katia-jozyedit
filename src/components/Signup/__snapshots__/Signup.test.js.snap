// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders the UI as expected 1`] = `
<main
  id="lp__main-flex"
>
  <div
    className="lp__logoContainer"
  >
    <a
      className="lp__lp-link"
      href="/"
      onClick={[Function]}
    >
      Katia
    </a>
  </div>
  <section
    className="Signup__container"
  >
    <h2
      className="Signup__header"
    >
      Signup
    </h2>
    <form
      className="Signup__form"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="Signup__email-div lp__input-div"
      >
        <label
          className="lp__label"
          htmlFor="Signup__email"
        >
          Email
        </label>
        <input
          className="lp__input"
          id="Signup__email"
          name="email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="Signup__display_name-div lp__input-div"
      >
        <label
          className="lp__label"
          htmlFor="Signup__display_name"
        >
          Display Name
        </label>
        <input
          className="lp__input"
          id="Signup__display_name"
          name="display_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="Signup__password-div lp__input-div"
      >
        <label
          className="lp__label"
          htmlFor="Signup__password"
        >
          Password
        </label>
        <input
          className="lp__input"
          id="Signup__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="Signup__confirm_password-div lp__input-div"
      >
        <label
          className="lp__label"
          htmlFor="Signup__confirm_password"
        >
          Confirm Password
        </label>
        <input
          className="lp__input"
          id="Signup__confirm_password"
          name="confirm_password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="blue-button blue-button-margin"
        type="submit"
      >
        Submit
      </button>
      <div
        className="Signup__submit-button-div"
      >
        <p
          className="Signup__or"
        >
          Already have an account? 
        </p>
        <a
          className="Signup__login-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </form>
  </section>
</main>
`;
